var/spack/repos/builtin/packages/variorum/package.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Variorum(CMakePackage):
+    """Variorum is a library providing vendor-neutral interfaces for
+       monitoring and controlling underlying hardware features.
+    """
+    homepage = "https://variorum.readthedocs.io"
+    git = "https://github.com/llnl/variorum.git"
+    url = "https://github.com/LLNL/variorum/archive/v0.1.0.tar.gz"
+
+    maintainers = ["slabasan", "rountree"]
+
+    version("0.1.0", tag="v0.1.0")
+
+    ############
+    # Variants #
+    ############
+    variant("shared", default=True, description="Build Variorum as shared lib")
+
+    # variants for language support
+    variant("python", default=True, description="Build Variorum Python support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=False, description="Build Variorum's documentation")
+    variant("log", default=False, description="Enable Variorum's logs")
+    variant("debug", default=False, description="Enable Variorum's debug")
+
+    ########################
+    # Package dependencies #
+    ########################
+    depends_on("cmake@2.8:", type="build")
+    depends_on("hwloc@1.11.9")
+
+    #########################
+    # Documentation related #
+    #########################
+    depends_on("py-sphinx", when="+python+docs", type="build")
+
+    root_cmakelists_dir = "../spack-src/src"
+
+    def cmake_args(self):
+        spec = self.spec
+        cmake_args = []
+
+        # if we have a static build, we need to avoid any of
+        # spack's default cmake settings related to rpaths
+        # (see: https://github.com/LLNL/spack/issues/2658)
+        if "+shared" in spec:
+            cmake_args.extend(std_cmake_args)
+        else:
+            for arg in std_cmake_args:
+                if arg.count("RPATH") == 0:
+                    cmake_args.append(arg)
```

Comment:
This won't work with the `CMakePackage` base class, as `std_cmake_args` is appended after your `cmake_args`. Is #2658 still an issue? Maybe you can comment on that issue and we can try to fix it in the base class.

Reply:
Nope, I was able to just append to the `cmake_args` if `+shared`.